backend-docker/pier/db/db.go

```diff
@@ -0,0 +1,551 @@
+package db
+
+import (
+	"bytes"
+	"database/sql"
+	"fmt"
+	"github.com/EUDAT-GEF/GEF/backend-docker/pier/internal/dckr"
+	_ "github.com/mattn/go-sqlite3"
+	"github.com/pborman/uuid"
+	"gopkg.in/gorp.v1"
+	"log"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const GefSrvLabelPrefix = "eudat.gef.service." // GefSrvLabelPrefix is the prefix identifying GEF related labels
+const GorpVersionColumn = "Revision"           // Column in a table used to keep an internal version number of GORP
+const SQLiteDataBasePath = "gef_db.bin"
+
+// VolumeID contains a docker volume ID
+type VolumeID dckr.VolumeID
+
+// Db is used to keep DbMap
+type Db struct{ gorp.DbMap }
+
+// Job stores the information about a service execution (used to serialize JSON)
+type Job struct {
+	ID           JobID
+	ServiceID    ServiceID
+	Input        string
+	Created      time.Time
+	State        *JobState
+	InputVolume  VolumeID
+	OutputVolume VolumeID
+	Tasks        []Task
+}
+
+// JobState keeps information about a job state
+type JobState struct {
+	Error  string
+	Status string
+	Code   int
+}
+
+// JobID exported
+type JobID string
+
+// JobTable stores the information about a service execution (used to store data in a database)
```

Comment:
This datatype (as all the other ones related to db) does not need to be public, it should be internal to this package (lower case).